feat: add support for TES v1.1.0

LakiG opened the pull request:
Fixes #11.

Added TES v1.1.0 `pydantic` model

LakiG committed:
feat: add support for TES v1.1.0

lbeckman314 commented:
This looks really good @LakiG! Any other updates needed before @uniqueg completes a review?

LakiG commented:
Thanks @lbeckman314. This PR will just have the new model and basic backward compatibility to the existing codebase.
I will add new tests (`filter` functionality) separately.

lbeckman314 commented:
Ah, understood! Would it work for me to rebase the other tests from #29 onto this PR for future development?

LakiG commented:
Yes, that would work.

uniqueg commented on compliance_suite/job_runner.py:
```diff
@@ -126,6 +126,24 @@ def tag_matcher(
+            True, if the versions match. Otherwise, false.
+        """
+
+        if self.version in versions:
+            return True
+        return False
```
Note that you could use Python's faux ternary operator here:

```python
True if self.version in versions else False
```

But that's just FYI, no need to change ;-)

uniqueg commented on compliance_suite/test_runner.py:
```diff
@@ -94,7 +95,10 @@ def validate_logic(
 
         try:
-            ENDPOINT_TO_MODEL[endpoint_model](**json_data)
+            pydantic_module: Any = importlib.import_module(
+                "compliance_suite.models.v" + self.version.replace('.', '_') + "_specs")
+            pydantic_model_class: Any = getattr(pydantic_module, ENDPOINT_TO_MODEL[endpoint_model])
+            pydantic_model_class(**json_data)  # JSON validation against Pydantic Model
```
Sweet! :)

uniqueg commented on unittests/test_job_runner.py:
```diff
@@ -34,34 +34,42 @@ class TestJobRunner(unittest.TestCase):
         """ Checks if generate summary functions runs successfully"""
 
-        job_runner_object = JobRunner('https://test.com/', 'v1.0', [])
+        job_runner_object = JobRunner('https://test.com/', '1.0.0', [])
```
I think it would really make sense to make use of parametrization in your tests here so that you can run checks against multiple versions. Same goes for all other tests in this and the next module.

Check https://docs.pytest.org/en/6.2.x/parametrize.html

uniqueg commented on unittests/test_job_runner.py:
```diff
@@ -34,34 +34,42 @@ class TestJobRunner(unittest.TestCase):
 
+    def test_version_matcher_success(self):
+        job_runner_object = JobRunner('https://test.com/', '1.0.0', ["tag"])
+        assert job_runner_object.version_matcher(["1.0.0", "1.1.0"]) is True
```
Again a good use case for parametrization

uniqueg commented on unittests/test_test_runner.py:
```diff
@@ -22,136 +22,143 @@ class TestTestRunner(unittest.TestCase):
-            {
-                "operation": "test",
-                "endpoint": "test"
+        for version in ["1.0.0", "1.1.0"]:
```
Again a good use case for parametrization. No need for the `for` loops then...

uniqueg reviewed (changes requested):
Cool stuff. I think the unit tests would really benefit from making use of parametrization, especially to make them more future proof when additional TES versions appear, or when generalizing the test runner for use with other specs (which likely use other versions).

uniqueg commented:
Very sorry for my very late review, been too busy! 🙈
However, this was the last issue in my backlog, so next ones will be _much_ faster!

LakiG committed:
feat: update unittests

LakiG committed:
feat: add unittest constants

LakiG commented:
No problem, thank you Alex for reviewing.
I have added parameterization, its quite helpful. I also did basic cleanup/refactoring, though it can be further improved.

uniqueg reviewed (approved):
Awesome!

However, before merging, let's first pull in @lbeckman314's test cases :)

lbeckman314 committed:
Feature/tesv1.1 (#32)

* Add tests for optional filetypes in inputs and outputs

* Add test for wildcards in task outputs

* Add test for backend_parameters flag in task resources

* Add v1.0/v1.1 compatible test for resources in task creation test

* Add test for streamable flag in tesInput

* Update title and description in streamable flag test

* Add test for ignore_error flag

* Update title and description in ignore_error flag test

* Add test for CANCELING state to delete cloud resources

- Add check for CANCELING state in functions test

* Update v0.1 task creation tests

- Remove extra executors
- Change executor images from ubuntu to alpine
- Update tags with specific test names
- Change default executor commands to `echo hello`
- Update version format to fit new versioning scheme
- Remove v1.0.0 from v1.1 tests

LakiG committed:
refactor: simplify polling state check

LakiG commented on tests/create_task_ignore_error.yml:
```diff
@@ -0,0 +1,52 @@
+            "command": [
+              "ERROR"
+            ],
+            "ignore_error": true
```
Should we add this test after the `filter` PR? As discussed, we can implement the array size check for the executor logs.

uniqueg replied:
Yes, I think that makes sense

LakiG committed:
feat: correct test files

LakiG committed:
feat: add backend positive & negative, inputs tests

LakiG committed:
feat: remove ignore_error test

LakiG committed:
feat: handle error response status, logger changes

uniqueg reviewed (approved):
The PR is too big for me to efficiently review. But from what I can see it looks good to me, and we have already discussed and resolved several issues. I'd say, let's merge it, and if we stumble on any bugs or unintended behaviors, we can create issues for those.

LakiG commented:
Sure. I have also reviewed it multiple times now. I think it should work.
Merging this.
